File: 2026-01-January-GFG-POTD/January 29 - Stream First Non-repeating/Solution.py
from collections import deque

class Solution:
    def firstNonRepeating(self, s):
        freq = [0] * 26
        q = deque()
        result = []

        for ch in s:
            # Update frequency
            freq[ord(ch) - ord('a')] += 1

            # Add current character to queue
            q.append(ch)

            # Remove repeating characters from front
            while q and freq[ord(q[0]) - ord('a')] > 1:
                q.popleft()

            # Append result for current prefix
            if q:
                result.append(q[0])
            else:
                result.append('#')

        return "".join(result)

File: 2026-01-January-GFG-POTD/January 29 - Stream First Non-repeating/Solution.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    string firstNonRepeating(string s) {
        vector<int> freq(26, 0);
        queue<char> q;
        string result = "";

        for (char ch : s) {
            // Update frequency
            freq[ch - 'a']++;

            // Add current character to queue
            q.push(ch);

            // Remove repeating characters from front
            while (!q.empty() && freq[q.front() - 'a'] > 1) {
                q.pop();
            }

            // Append result for current prefix
            if (q.empty()) {
                result.push_back('#');
            } else {
                result.push_back(q.front());
            }
        }

        return result;
    }
};
